From 976996e3f336bcdadc6e8500f477d1ae3ff82388 Mon Sep 17 00:00:00 2001
From: Byron McCollum <byron.mccollum@rackspace.com>
Date: Tue, 10 Nov 2015 07:38:28 -0800
Subject: [PATCH] Wait for container SSH

This adds a wait condition when creating the containers to ensure
that the containers are created correctly and that ssh is available
within them before actually continuing with the installation process

Change-Id: I78926651ea7a0cdfd6d903c590b421448264a4fb
---

diff --git a/playbooks/lxc-containers-create.yml b/playbooks/lxc-containers-create.yml
index 0b5110a..daa1355 100644
--- a/playbooks/lxc-containers-create.yml
+++ b/playbooks/lxc-containers-create.yml
@@ -20,6 +20,16 @@
   user: root
   roles:
     - { role: "lxc_container_create", tags: [ "lxc-container-create" ] }
+  post_tasks:
+    - name: Wait for ssh to be available
+      local_action:
+        module: wait_for
+        port: "{{ ansible_ssh_port | default('22') }}"
+        host: "{{ ansible_ssh_host | default(inventory_hostname) }}"
+        search_regex: OpenSSH
+        delay: 1
+      tags:
+        - lxc-container-wait-for-ssh
   vars:
     ansible_hostname: "{{ container_name }}"
     is_metal: "{{ properties.is_metal|default(false) }}"
